pkg/scheduler/framework/plugins/noderesources/resource_allocation.go

```diff
@@ -51,18 +45,17 @@ func (r *resourceAllocationScorer) score(
 	if node == nil {
 		return 0, framework.NewStatus(framework.Error, "node not found")
 	}
-	if len(r.resourceToWeightMap) == 0 {
+	// resources not set, nothing scheduled,
+	if len(r.resources) == 0 {
 		return 0, framework.NewStatus(framework.Error, "resources not found")
 	}
 
-	requested := make(resourceToValueMap)
-	allocatable := make(resourceToValueMap)
-	for resource := range r.resourceToWeightMap {
-		alloc, req := r.calculateResourceAllocatableRequest(nodeInfo, pod, resource)
-		if alloc != 0 {
-			// Only fill the extended resource entry when it's non-zero.
-			allocatable[resource], requested[resource] = alloc, req
-		}
+	requested := make([]int64, 0, len(r.resources))
+	allocatable := make([]int64, 0, len(r.resources))
+	for i := range r.resources {
+		alloc, req := r.calculateResourceAllocatableRequest(nodeInfo, pod, v1.ResourceName(r.resources[i].Name))
```

Comment:
Isn't it simpler to keep: `if alloc == 0 { continue }`?

after this point, the names of the resources are not important. each index of requested and allocatable are already guaranteed to be for the same resource.

Reply:
Yes.

And I think the length of  `requested` and `allocatable` equal to `len(r.resources)`, so
```
requested := make([]int64, len(r.resources))
allocatable := make([]int64, len(r.resources))
```